src/lib/lightning/eclair/eclairApi.ts

```diff
@@ -40,3 +43,21 @@ export const httpPost = async <T>(
 ): Promise<T> => {
   return request<T>(node, 'POST', path, body);
 };
+
+const setupConfig = (eclairNode: EclairNode): Options => {
+  const base64auth = Buffer.from(`:${eclairCredentials.pass}`).toString('base64');
+  const config = {
+    url: `127.0.0.1:${eclairNode.ports.rest}`,
+    headers: {
+      Authorization: `Basic ${base64auth}`,
+    },
+  };
+  return config;
+};
+
+export const setupListener = (eclairNode: EclairNode): EclairWebSocket => {
+  const eclairConfig = setupConfig(eclairNode);
+  const eclairTs = new EclairTs(eclairConfig);
+  const listener = eclairTs.listen();
```

Comment:
I feel like adding the `eclair-ts` package just to call this one method is a bit much, especially since that package is no longer maintained. It hasn't been updated since Eclair v0.3.0. 

I would suggest to just implement the relevant code we need here in this file. You could just have a `listen` function that returns the WebSocket.

Reply:
Good suggestion. Replaced with only the function it needs